elastic/kibana
# [Lens] Leverage original http request error

Fixes https://github.com/elastic/kibana/issues/77360

This PR makes sure the underlying http request error is propagated to the expression error handler and uses the underlying ES information to show a more meaningful error message:
<img width="1130" alt="Screenshot 2020-10-07 at 12 05 20" src="https://user-images.githubusercontent.com/1508364/95317475-89cd1400-0895-11eb-8cbc-b6bb438485eb.png">

Full list of changes:
* `src/plugins/expressions/common/util/create_error.ts` If an error is thrown in a sub expression (as it is the case for Lens), the original error was not propagated correctly but discarded because an `ExpressionValueError['error']` is not a real JS `Error`. This makes sure to always keep the underlying real error object even if it's wrapped multiple times.
* `src/plugins/expressions/public/react_expression_renderer.tsx`: Only the error message was passed to the expression renderer react component callback - this PR adds a second parameter for the whole error
* `src/plugins/expressions/public/types/index.ts`: Add `original` to the type of this error - not sure about the type structure here, but it definitely ends up being defined at that level as well
* In Lens: Add a helper for pulling the interesting pieces out of the original error and use it in the workspace panel and in the embeddable.

## Testing

* Create a Lens dashboard based on an index
* Create a user without read rights on that index
* Access the dashboard logged in as that user
* Error messages should be helpful
